maths/double_factorial_recursive.py

```diff
@@ -0,0 +1,31 @@
+def double_factorial(n: int) -> int:
```

Comment:
I think it is not necessary because it is particularly used in various examples like maths / factorial_iterative.py, maths / perfect_cube.py, backtracking / all_combinations.py and others more.